numba/typed/listobject.py

```diff
@@ -416,21 +414,26 @@ def _list_length(typingctx, l):
 
     Returns the length of the list.
     """
-    resty = types.intp
-    sig = resty(l)
+    sig = types.intp(l)
+    ll_list_type = cgutils.voidptr_t
```

Comment:
I am not sure why this is redefined here? There is an original definition at the top of the file. Is this re-definition required to ensure performance?

Reply:
It's because I wrote all the fixes outside of the actual compiler and pasted them in, there's probably duplicates, will fix.